### DISCUSSION TITLE
Main Contributor's Workflow

### DISCUSSION THREAD

**Discussion by djbrown** [GnuCash-Pocket/gnucash-android#207] [Team Posts]
Hey @pnemonic78 😊 Thank you very much for all the work you have put into this project! I'd like to ask and maybe discuss some more "background" topics to better understand your whole workflow.

### Pull Request Reviews
I had noticed subsequent updates (commits) on already reviewed pull requests, and I was always wondering why that was.
Yesterday I took some time again to look over pull requests with requested reviews. Most of them got merged today. Good thing is GitHub meanwhile supports incremental reviews, so it keeps track of previously viewed changes and only shows outdated files on reviews for updated pull requests.
I just wanted you to know I was reluctant to review updated pull requests in the past because I didn't want to look over the same changes twice.
And also please keep requesting reviews when you want me to look over pull requests 👍🏽
Might still take some time, depending on my spare time, but I'll always try to prioritize those requests over others.

### Upstream Fork Sync
For an open source project I had always assumed the upstream repo to be the "official" repo e.g. for ci and release mechanics, and that personal forks would be just for temporary work on contributions.
You are by far the most active developer on this project. Do I understand correctly, that your personal fork is your "official" repo, not only for development but also ci and especially code base for release? I know you try to keep this org repo up to date as much and often as possible, but then still the actual "upstream" would be your repo. Is my impression skewed or is there something to it?
We shouldn't call any shots from that yet, I'm just asking for your mental image of this.

### App Store Release
For this one I'd like to know how you build and deploy to the app store. I try not to go too deep into detail but I'm pretty curious.
Are there many manual steps on a (certain) local or remote machine and are they (publicly) documented?
Or is it rather automated via script or service/platform?
Is this process pretty stable, or does it vary from release to release?
Could someone else do the release instead of you?
How much time does one release take you on average from start to finish?
How often do you release on average?
Do you release to other app stores besides Google Play?

### Support Channels
Over which ways do you receive feedback, other than issues from this org repo?
Which of those are the most important ones to you? Which ones would you prefer in the future?
e.g. your personal repo, separate service/platform (Jira/BugZilla/Zendesk/UserVoice..), Google Play store comments, direct message/email, ...
And also where do you gather error stack traces from, which you paste into bug issue descriptions?
Do you have a personal Firebase Crashlytics dashboard, or other platform, or do people send them along with bug reports?

With all these questions I hope to understand better your additional project efforts on top of all the code contributions 😉

**Comment by pnemonic78** [comment]
re Pull Request Reviews
A pull request is updated because of the "force push" because of conflicts from other merges.

re Upstream Fork Sync
Yes, my fork is where I release from because I have many branches that are WIP (not merged to upstream).

re App Store Release
After merging relevant branches to "main", run gradle task to build app bundle, then manually publish to play store.
So far only play store.

re Support Channels
4 main ways: crashlytics, github issue tracker, play store feedback. and even some emails

**Comment by rogeryanda** [comment]
I am not sure where to report a problem with the calculation of net worth. The value I  see is more than double the total assets. Equity is also being counted after it was allocated.

**Comment by eozalp** [comment]
- This post comes here because can't create on other sections -

I think using bottom menu will be more useful because: 

1. Ease of Reach:
On mobile devices, users naturally interact with the lower part of the screen using their thumbs. A bottom menu puts key actions within easy reach.


2. Faster Navigation:
Frequent or essential sections (e.g., favorites, settings, all, settings) are just a tap away — no need to open side drawers or scroll.


3. Better UX for PWA/Apps:
Bottom nav is the standard for mobile apps (especially iOS and Android guidelines), so users are already familiar with it.


4. Always Visible (Optional):
If designed properly, it can stay visible while navigating, making transitions smoother and consistent.


5. Cleaner UI:
It removes the need for a top-heavy layout or hamburger menu, leading to a more modern, minimal design.

**Comment by FlavioB79** [comment]
I've got to ask in this thread as well: how can I start a new discussion?
